### PR TITLE
Remove FixedCountNodeAllocatorService

### PR DESCRIPTION
**Pull request by losipiuk**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by linzebing — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/execution/scheduler/NodeSchedulerConfig.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import static java.util.Locale.ENGLISH;
 
 @DefunctConfig({
+        "node-scheduler.allocator-type",
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/NodeSchedulerConfig.java:30]: > Deprecate node-scheduler.allocator-type config property


Deprecate -> Remove

File: core/trino-main/src/test/java/io/trino/execution/TestNodeSchedulerConfig.java [hunk 2]
```diff
@@ -60,7 +59,6 @@ public void testExplicitPropertyMappings()
                 .put("node-scheduler.splits-balancing-policy", "node")
                 .put("node-scheduler.optimized-local-scheduling", "false")
                 .put("node-scheduler.allowed-no-matching-node-period", "1m")
-                .put("node-scheduler.allocator-type", "fixed_count")
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/execution/TestNodeSchedulerConfig.java:63]: Does TestNodeSchedulerConfig pass after first commit?

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/execution/TestNodeSchedulerConfig.java (hunk 2 above)
Comment (losipiuk) [reply core/trino-main/src/test/java/io/trino/execution/TestNodeSchedulerConfig.java:63]: It does :)

**Comment by losipiuk** [comment]
rebased

**Comment by losipiuk** [comment]
CI : https://github.com/trinodb/trino/issues/14277

**Pull request merged**